### PR TITLE
Add flash simulator support

### PR DESCRIPTION
**Pull request by dsemenets**

### CODE REVIEW

**Comment by oleksiimoisieiev** [comment]
LGTM.

**Comment by oleksiimoisieiev** [comment]
Reviewed-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>

File: west.yml
Comment (xakep-amatop) [inline west.yml:27]: (question) Do we need to update this revision every time when we or somebody else add changes related to littlefs into `zephyrproject-rtos`? why can't we use some branch here ?

File: boards/xenvm-qemu.overlay
Comment (xakep-amatop) [inline boards/xenvm-qemu.overlay:29]: some notes from Zephyr documentation:
```
Earlier versions of Zephyr made frequent use of the label property, which is distinct from the 
tandard node label. Use of the label property in new devicetree bindings, as well as use of the
DT_LABEL macro in new code, are actively discouraged. Label properties continue to persist
for historical reasons in some existing bindings and overlays, but should not be used in new
bindings or device implementations.
```

moreover, I have seen warnings form `west` that `label` property is a deprecated during compilation of a project where I had DTS labels.

**Comment by xakep-amatop** [comment]
review finished,
pls, answer the question and avoid `label` property in overlay if it possible

**Review by xakep-amatop — CHANGES_REQUESTED** [review]

File: west.yml
Comment (dsemenets) [reply west.yml:27]: This module is not in our repository. I don't think we have to make fork for this repository until we make changes here.

File: boards/xenvm-qemu.overlay
Comment (dsemenets) [reply boards/xenvm-qemu.overlay:29]: This is partition on flash simulator and added here for test purpose on qemu. On product branch we will not use it.

File: west.yml
Comment (xakep-amatop) [reply west.yml:27]: why can't you put name of branch instead of commit hash even without fork?

File: boards/xenvm-qemu.overlay
Comment (xakep-amatop) [reply boards/xenvm-qemu.overlay:29]: could you please add todo that we have to delete this chunk of code before merging it to product branch?

File: west.yml
Comment (oleksiimoisieiev) [reply west.yml:27]: Using exact commit is common approach. This ensures us that no update in littefs module will break our code. When we need newer version - we should just bump up the later commit sha

File: boards/xenvm-qemu.overlay
Comment (oleksiimoisieiev) [reply boards/xenvm-qemu.overlay:29]: I think whole qemu-xenvm board is for the testing purposes. But it should be merged so other functionality can be tested based on this.

File: west.yml
Comment (xakep-amatop) [reply west.yml:27]: so, is our functionality written in such way that further updating of the liitlefs and fixing bugs by maintainers can break our functionality?.. updating some functionality should not break backward compatibility and I am sure that maintainers of the littlefs knew about that, at least, we can stay on some release branch where everything is stable

File: boards/xenvm-qemu.overlay
Comment (xakep-amatop) [reply boards/xenvm-qemu.overlay:29]: I afraid that we can forget about deleting this code in the future
but ok, I got your point, thanks

File: west.yml
Comment (oleksiimoisieiev) [reply west.yml:27]: > updating some functionality should not break backward compatibility and I am sure that maintainers of the littlefs knew about that,

I'm not so sure about that. That's why this is a common way to use tags, not branches for the releases (linux kernel for example) and all references to the exact kernel is based on tag, not branch. Basing on branch, even release branch is bad practice because we can't be sure that your build = customer build. And what if some bug appeared in littlefs with newly pushed commits? Are you ready to spend time fixing it?

Comment (xakep-amatop) [reply west.yml:27]: so, do you able to use tag instead of hash here?

Comment (xakep-amatop) [reply west.yml:27]: > And what if some bug appeared in littlefs with newly pushed commits? Are you ready to spend time fixing it?

Actually if you stay on stable tag or released branch, I think you should not care about that
this is the my point

Comment (oleksiimoisieiev) [reply west.yml:27]: I don't know if I'm able to use tag here. I haven't looked on this.

> Actually if you stay on stable tag or released branch, I think you should not care about that
this is the my point

I don't accept branch, even release branch based on branch and tag definition in GIT.

Comment (dsemenets) [reply west.yml:27]: This is not our repository. We can't set any tags or branches. I can't see any reason to fork this repo for chenging this module.

Comment (xakep-amatop) [reply west.yml:27]: @dsemenets  got it, thanks for the answer

**Pull request closed without merging**
